std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(is(T : const char[]) || is(T : const wchar[]) || is(T : const dchar[]))
```

Comment:
@BBasile, @lionello wants to add support for static arrays of characters, which `isSomeString` doesn't allow. The shorter version would be:
```
import std.traits : isSomeChar;
else static if (is(T : E[], E) && isSomeChar!E)
```